docs/ImpactAnalysis.md

```diff
@@ -28,12 +28,130 @@ This document is the first iteration of a high-level design document for the Lei
+
+### Client interfaces
+
+There are two major [client interfaces](https://cardano-scaling.github.io/cardano-blueprint/client/index.html) served by Cardano node: the node-to-client (N2C) protocol and the [UTxO-RPC](https://utxorpc.org/) interface. The N2C protocol follows the design used between nodes on the peer-to-peer network, but using a different set of mini-protocols that allow clients to query the ledger state, submit transactions, and conveniently follow the chain. The UTxO-RPC interface is a gRPC interface for UTxO blockchains that allows clients to query chain data and submit transactions. Both interfaces are abstracting away details of how consensus is achieved and therefore are only marginally impacted by the Leios consensus change.
+
+> [!WARNING]
+> TODO: Update cardano-blueprint for LocalChainSync and link it here or find another non-ogmios reference
+
+Both client interfaces provide access to the chain as a sequence of blocks: N2C via the `LocalChainSync` (see [ogmios API reference](https://ogmios.dev/mini-protocols/local-chain-sync/)) and UTxO-RPC via the [Sync Module](https://utxorpc.org/sync/intro/). As the Leios upgrade will change the block format, those APIs will need to be updated to reflect the new block structure. However, the fundamental concept of a block containing transactions remains unchanged and a backwards compatible representation of blocks is possible. For example, the `LocalChainSync` protocol is serving currently Praos blocks, whose body is basically a list of transactions (see the [Conway CDDL](https://github.com/IntersectMBO/cardano-ledger/blob/master/eras/conway/impl/cddl-files/conway.cddl)). With Leios, this would change to either a list of transactions or a certificate. However, a backwards compatible way would be to "inline" the transactions referenced via an EB in the block body, so that clients can continue to consume transactions directly from the block body. Two things need to be considered with this approach:
```

Comment:
> a backwards compatible representation of blocks is possible

Is there idea here that the N2C ChainSync mini-protocol could simply serve a CertRB by inlining the certified EB into it as if it were a _huge_ TxRB?

I could imagine how that would be useful in some scenarios, but I also expect it would cause some (perhaps acceptable) confusion. Such a faked-up mega-RB will at least violate the per-RB limits on tx measurement dimensions (bytes, ExUnits, refscript bytes) and won't have the size nor the hash claimed by its Praos header.

An alternative is for N2C ChainSync so simply serve `(Maybe EB, RB)`. We could provide the `fakeUpMegaRB :: (Maybe EB, RB) -> RB` function (in a few popular languages) for developers to use as short-term integration hack, maybe?

Reply:
Ah, I see now that I jumped the gun writing this comment :D

Reply:
Yeah.. exactly below there is this concern :D At this point we should be fine with pointing this out and see how client interfaces fall into place (how would the ideal Leios client API look like?).